lux-lib/src/operations/run_lua.rs

```diff
@@ -33,39 +38,67 @@ pub enum RunLuaError {
     },
     #[error(transparent)]
     Paths(#[from] PathsError),
+
+    #[error(transparent)]
+    Tree(#[from] TreeError),
 }
 
-pub async fn run_lua(
-    root: &Path,
-    tree: &Tree,
+#[derive(Builder)]
+pub struct RunLuaBuilder<'a> {
+    root: &'a Path,
+    tree: &'a Tree,
+    config: &'a Config,
     lua_cmd: LuaBinary,
```

Comment:
Sorry, somehow this one got lost in my previous review.

issue: This is not consistent with how we use `bon::Builder`. You can see an example in `lux-lib::operations::test`.

```suggestion
#[derive(Builder)]
#[builder(start_fn = new, finish_fn(name = _run, vis = ""))]
pub struct RunLua<'a> {
    root: &'a Path,
    tree: &'a Tree,
    config: &'a Config,
    lua_cmd: LuaBinary,
```

The builder should then be called in `lux-cli::run_lua` like this:

```rust
operations::RunLua::new()
        .root(&root)
        .tree(&tree)
        .config(&config)
        .lua_cmd(lua_cmd)
        .args(args)
        .prepend_test_paths(run_lua.prepend_test_paths)
        .prepend_build_paths(run_lua.prepend_build_paths)
        .run_lua()
        .await?;